### PR TITLE
[WIP] Add action to update all installed fastlane tools

### PR DESCRIPTION
**Pull request by milch**
This action simply looks up all the fastlane.tools you have installed and updates them. If any of the fastlane.tools were updated, fastlane is restarted. 

Recommended usage would be in a before_all block in the Fastfile. 

This is WIP, because the following tasks still need to be done:
- [x] Update Actions.md documentation to explain the usage
- [ ] ~~Add second action that helps with setting up ruby gems so we don't need to sudo every time~~
- [x] Handle the situation where we would need root privileges correctly
- [x] Test parameter handling (e.g. updating fastlane + deliver only)
- [x] Don't update major versions (e.g. 1.0 to 1.1 is fine, 1.5 to 2.0 is not)

### CODE REVIEW

File: lib/fastlane/actions/update_fastlane.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+require 'rubygems/commands/update_command'
+require "open3"
+
+module Fastlane
+  module Actions
+    # Makes sure fastlane tools are up-to-date when running fastlane 
+    class UpdateFastlaneAction < Action
+
+      ALL_TOOLS = [
+        "fastlane",
```
Comment (KrauseFx) [inline lib/fastlane/actions/update_fastlane.rb]: You forget the good old https://rubygems.org/gems/credentials_manager :cry:

**Comment by milch** [comment]
Did a rebase by accident, so got a lot of commits from feature/config into the commit list. Files changed shows correct files though

**Comment by KrauseFx** [comment]
Great, is this ready to merge? Should I test it? :+1:

**Comment by milch** [comment]
Yeah, you can test it now, everything should work correctly :+1:

**Comment by KrauseFx** [comment]
Thank you! :rocket:

**Pull request merged**